### PR TITLE
feat: clean shutdown simulation on sigterm

### LINKED ISSUE #58: Capture sigterm and trigger shutdown

### PR DESCRIPTION
**Pull request by okjodom**
capture sigterm from `sim-cli` and issue simulation shutdown. this allows us to complete tasks like report writing, before simulation exit

fixes #58

### CODE REVIEW

File: sim-lib/src/lib.rs [hunk 1]
```diff
@@ -292,10 +302,11 @@ impl Simulation {
     async fn generate_activity(
         &self,
         executed_actions: Sender<ActionOutcome>,
-        shutdown: Trigger,
-        listener: Listener,
     ) -> Result<JoinSet<()>, SimulationError> {
         let mut set = JoinSet::new();
+        let shutdown = self.shutdown_trigger.clone();
+        let listener = self.shutdown_listener.clone();
```
Comment (carlaKC) [inline sim-lib/src/lib.rs:308]: These aren't inline with `spawn` because of `self` escaping the borrow, right?

File: sim-cli/src/main.rs [hunk 2]
```diff
@@ -44,7 +44,14 @@ async fn main() -> anyhow::Result<()> {
         clients.insert(node.id, Arc::new(Mutex::new(lnd)));
     }
 
-    let sim = Simulation::new(clients, activity);
+    let sim = Arc::new(Simulation::new(clients, activity));
+    let sim2 = sim.clone();
```
Comment (carlaKC) [inline sim-cli/src/main.rs:48]: Not my _favorite_ var name 🙈

**Review by carlaKC — APPROVED** [review]
tACK ✅

File: sim-lib/src/lib.rs (hunk 1 above)
Comment (okjodom) [reply sim-lib/src/lib.rs:308]: correct. we can never move `self` into the spawn

**Pull request merged**